### ISSUE TITLE
OOM crash due to code bug in appendNextSiblings function (invoked when used with CSS query for siblings)

### ISSUE THREAD

**Issue by MikePikanin** [fb55/css-select#655]
# Basic info
* Node version: 14.16.3
* css-select version: 4.1.3, but I am looking at the current latest version 4.2.0 and it also applies there.
* cheerio (in example): v1.0.0-rc.10

# Description
In the function https://github.com/fb55/css-select/blob/0e526af6fe2e000588431b5d4e8038bfc3e9bfdb/src/index.ts#L112 there is an array of elements which is being iterated and in each cycle more elements are being added to it, this leads to OOM crashes where there  is enough siblings (e.g. 50).
Error: `FATAL ERROR: invalid array length Allocation failed - JavaScript heap out of memory`

# How to reproduce
```let cheerio = require('cheerio');
let html = "<html><div class="parent"></div></html>";
let $ = cheerio.load(html);
for (let i=0; i<50; i++)
{
    $('.parent').append('<div class="child">1</div>');
}
$('.child').find('+.child');
```
# Suggested fix:
Get array length before entering the loop and use it in loop condition.

**Comment by fb55** [comment]
Great catch, thanks for the detailed report! I've fixed this in #656 & will get a new release out shortly.

**Comment by fb55** [comment]
Fixed in `4.2.1`!
